File: java-sandbox/java-maven-tests/src/main/java/com/example/sandbox/Strings.java
package com.example.sandbox;

import java.util.List;

/** String helpers used by the example tests. */
public final class Strings {

    private Strings() {
    }

    public static String sayHi(String name) {
        return "Hi, " + name + "!";
    }

    public static String slugify(String text) {
        return text.toLowerCase().replace(' ', '-');
    }

    public static String reverse(String text) {
        return new StringBuilder(text).reverse().toString();
    }

    public static boolean isPalindrome(String text) {
        return text.equals(reverse(text));
    }

    public static List<String> words(String sentence) {
        if (sentence.isBlank()) {
            return List.of();
        }
        return List.of(sentence.trim().split("\\s+"));
    }
}

File: csharp-sandbox/cs-nunit-tests/src/Strings.cs
namespace Sandbox;

/// <summary>String helpers used by the example tests.</summary>
public static class Strings
{
    public static string SayHi(string name) => $"Hi, {name}!";

    public static string Slugify(string text) => text.ToLowerInvariant().Replace(' ', '-');

    public static string Reverse(string text)
    {
        var chars = text.ToCharArray();
        Array.Reverse(chars);
        return new string(chars);
    }

    public static bool IsPalindrome(string text) => text == Reverse(text);

    public static IReadOnlyList<string> Words(string sentence) =>
        string.IsNullOrWhiteSpace(sentence)
            ? Array.Empty<string>()
            : sentence.Trim().Split((char[]?)null, StringSplitOptions.RemoveEmptyEntries);
}
